I16 urban vs rural communities

solveigvandervegt opened the pull request:
pull request for the 2 communities model to be added to the como-models package.

Some issues remain around loading the excel files and I can therefor also not check if all the tests actually pass.

solveigvandervegt committed:
Added first U vs R model

Added first model with documentation and vignette. This one assumes people stay in their population. Need to edit vignette, just wanted to check that the code works. Need to add second model too.

solveigvandervegt committed:
Implemented second version urban vs. rural

Created a class for the urban vs. rural model where migration between communities is allowed, but only infection within a community is possible. Also edited the vignette to show a comparison of the two models.

It's all a bit rough and ready at the moment cause I just wanted to make sure the models worked and to see how they compare but the there will most certainly be errors in the documentation and the vignette. Also, there are no tests yet for either of these new urban vs. rural classes.

solveigvandervegt committed:
update vignette

added some thoughts on urban vs rural model comparison

solveigvandervegt committed:
Add model comparison

Add a .R file to compare the two approaches to the urban vs. rural model without having to go through the vignette.

solveigvandervegt committed:
Trying to figure out contact data

Donwloaded urban and rural contact matrices

Added some code to explore them before comparing UvsY and normal SEIR model

solveigvandervegt committed:
Edited nomig rural vs urban model

Added in data from the world bank on the percent rural population for each country.

Edited how parameters are computed.

solveigvandervegt committed:
Add World bank data

Add excel sheet with world bank data on rural populations in different countries to Data folder

solveigvandervegt committed:
Change death rate to be Tunisia specific

solveigvandervegt committed:
removed some redundant plots

Also played around with some of the parameters and countries so those changed a little bit but will be user-specified anyway

solveigvandervegt committed:
Changed contact matrix processing

Now sum contacts in one age group and then average across age groups, use rural:urban ratio for scaling.

Before I just took the mean of the whole matrix, but I don't think that is right.

Still not entirely sure how to incorporate this into model tbh

solveigvandervegt committed:
Incorporating contact matrices

Trying very hard, not working at all

solveigvandervegt committed:
Add demog data and fixed ODEs

Finally figured out how to properly write the ODEs for normalized populations.

Add demographic data for 5 year age groups to properly take the weighted sum of the contact matrices and obtain the total number of contacts in a community

solveigvandervegt committed:
Updated ODEs

I think doing it this way takes better into account the mixing of populations and has us a pick a connectedness parameter instead of guess how many contacts are between populations.

solveigvandervegt committed:
Changed ODEs (again) and added exploration of C

solveigvandervegt committed:
Summed communities and deleted unnecessary files

Add the option to plot the output summed over communities to allow for easier comparison to the basic SEIR model.

solveigvandervegt committed:
Add file to compare R0

I am not sure if the computations are correct because the R0 for the basic model doesn't match the urban vs rural model for C=1 and I think it should at least be close. To be continued

solveigvandervegt committed:
add R0

add file to compute R0 and fixed bug in computation of average number on contacts

solveigvandervegt committed:
Added comparisons

edited plot to include an arrow indicating whether R0 for the urban vs rural model is higher or lower than for the SEIRD model

solveigvandervegt committed:
Added R0(C)

Added a plot of R0 as a function of C for different countries

solveigvandervegt committed:
Split scripts up

Separated some code for clarity

solveigvandervegt committed:
Add a class with time-dependent C

Add a new class where C is set as a time-dependent series (discrete time).

solveigvandervegt committed:
Clean up and vignette

Removed all the files that don't need to go into the package

Wrote vignette

solveigvandervegt committed:
Made test file

Having problems with finding the excel files both in the vignette and in the test file.

solveigvandervegt committed:
Fixed problems with loading data

converted excel files to Rdata files.

Fixed vignette and tests so they now load all data correctly.

solveigvandervegt committed:
Fixed warnings and notes

Fixed some of the warnings and notes when checking the code. Still have some remaining but not too sure what they want or how to fix it.

solveigvandervegt committed:
made data files smaller

solveigvandervegt committed:
Merge branch 'main' into i16-urban-vs-rural-communities

rccreswell commented on DESCRIPTION:
```diff
@@ -19,6 +19,9 @@ Suggests:
     deSolve,
+    matlib,
+    readxl,
+    gridExtra
```
```suggestion
    gridExtra,
```

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+library(reshape2)
+library(gridExtra)
+library(readxl)
+library(viridis)
```
Do we need to add viridis to Imports in DESCRIPTION?

solveigvandervegt replied:
yes, we do, thanks for the spot :)

rccreswell commented on DESCRIPTION:
```diff
@@ -19,6 +19,9 @@ Suggests:
 Imports: 
     deSolve,
+    matlib,
+    readxl,
```
Since you converted the excel files, do we still need readxl?

solveigvandervegt replied:
probably not, no, I'll take it out!

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+library(comomodels)
+library(ggplot2)
+library(glue)
+library(readxl)
```
```suggestion
```

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+library(deSolve)
+library(reshape2)
+library(gridExtra)
+library(readxl)
```
```suggestion
```
Assuming we don't need these anymore

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+#' @slot transmission_parameters list of values for transmission parameters
+#'       (double).
+#' @slot contact_matrices list of two contact matrices, one for urban communties
+#'       and one for rural communities, normalized to population size (double).
```
Not 100% sure what you mean by "normalized to population size" for the contact matrices

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+      stop("Contact matrices must be of type double.")
+    }
+
+    # check that vectors are normalized: loosing some precision in division
```
```suggestion
    # check that vectors are normalized: losing some precision in division
```

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+    # fraction of the population that is urban
+    f_urban <- sum(country_demog(object)$urban)
+    #fraction of the population that is rural
+    f_rural <- sum(country_demog(object)$rural)
```
It's great that the class provides `country_demog<-` and `contact_matrices<-` and the computations here, as they make it very easy to work with real data. However, I would strongly favor adding an option so that users can alternatively provide the values of N_U, N_Y, phi_U, and phi_Y directly, in case they want to run simulations with particular values in mind. 
We can probably achieve this using the same methods, but either in the docstrings or the vignette we should tell the user exactly what inputs to send to those methods if they just want to provide 4 numbers N_U, N_Y, phi_U, phi_Y.

solveigvandervegt replied:
Done!

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+  "transmission_parameters<-",
+  function(object, value) {
+    standardGeneric("transmission_parameters<-")
+  })
```
I think that these setGenerics will cause problems for the other classes. 
Based on https://github.com/Como-DTC-Collaboration/como-models/pull/5/commits/2cafce9eb38b9865f8144569b7e119b56eef2d37
https://github.com/Como-DTC-Collaboration/como-models/issues/21

Can you check that if you remove the setGeneric for transmission_parameters, run, and so forth, keeping only those for methods that are not included in any of the other classes, your model will still work?

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+  })
+
+#----------------------------------------------------------------------------
+#' Returns the value of R0 for the model with the specified parameter values.
```
Somewhere in the comments for this method, let's spell out that by R0 we mean "basic reproduction number"

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,694 @@
+      # convenience parameter
+      K <- ((1-f_rural)* NU + f_rural*NY)
+      
+      matF <- matrix(data = c(0, 0, b*S0U*(K*C + NU/(1-f_rural)*(1-C)),              b*S0U*K*C,
```
The arithmetic is quite complicated here. While there's no need to repeat the entire method if we cite it, can we:
- Wherever is typical for R code, name the same paper you have in the vignette
- Include a brief comment saying what F and V mean, and that matF and matV are where you are putting the derivatives
- Indicate what ordering you have adopted of the state variables (I_U, E_U, ...) for your matrices

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+
+## The mathematical model
+### The equations
+We base our model of connected communities on the model presented by Tun et al. [-@Tun21]. We designate the urban community $U$ and the rural community $Y$. Following the same naming pattern for variables and parameters as used for the basic SEIRD model, the ordinary differential equations describing how the populations evolve are as follows, where variable $X_Z$ designates the population of indi ...
```
```suggestion
We base our model of connected communities on the model presented by Tun et al. [-@Tun21]. We designate the urban community $U$ and the rural community $Y$. Following the same naming pattern for variables and parameters as used for the basic SEIRD model, the ordinary differential equations describing how the populations evolve are as follows, where variable $X_Z$  (for $X \in \{S,E,I,R\}, Z \in \{U, Y\}$) designates the population of individuals with infectedness status $X$ in community $Z$,
```
Something like this could be nice so the new variables don't confuse anyone

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+\end{split}
+\end{equation}
+
+\noindent In the above equations, $\phi_U$ is the fraction of the population that lives in an urban environment, $\phi_Y$ is the fraction of the population that lives in a rural environment, $N_U$ is the average number of contacts that an urban individual has in a day (normalized over the whole population), and $N_Y$ is the average number of contacts that a rural individual has in a day (normaliz ...
```
Probably related to another comment---"normalize" is a bit vague, if you could make these definitions slightly more specific (maybe including an equation if necessary), that would be helpful

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+
+\noindent In the above equations, $\phi_U$ is the fraction of the population that lives in an urban environment, $\phi_Y$ is the fraction of the population that lives in a rural environment, $N_U$ is the average number of contacts that an urban individual has in a day (normalized over the whole population), and $N_Y$ is the average number of contacts that a rural individual has in a day (normaliz ...
+
+The difference between this model and the basic SEIRD model is the computation of the number of infectious contacts that a susceptible individual has. This is reflected in the equations for $S_Z$ and $E_Z$. Individuals in either community can either come into contact with people from their own community or from the other community. The extent to which the later happens is determined by the connec ...
```
```suggestion
The difference between this model and the basic SEIRD model is the computation of the number of infectious contacts that a susceptible individual has. This is reflected in the equations for $S_Z$ and $E_Z$. Individuals in either community can either come into contact with people from their own community or from the other community. The extent to which the latter happens is determined by the connectedness parameter $C$. If the communities are completely separate, i.e. $C=0$, the probability that a susceptible individual in community $Z$ interacts with an infectious individual is the number of contacts that someone in that community has, $N_Z$, multiplied by the probability that a contact is infectious, $I_Z/\phi_Z$. If the communities are well-mixed with each other, the probability is again the number of contacts multiplied by the probability that a contact is infectious, although now the number of contacts that an individual has is averaged over the communities, so $\phi_U N_U + \phi_Y N_Y$, and the individual can encounter people from either community so the probability of an infectious contact is $I_U+I_Y$. The extent to which the communities are connected, designated by $C$, determines to which extent the two extreme contact patterns play a role, and thus we obtain the weighted average that appears in the equations above which designates the total probability of a susceptible individual in $Z$ having an infectious contact: $(I_U+I_Y) (\phi_U N_U + \phi_Y N_Y) C + \frac{I_Z}{\phi_Z}N_Z(1-C)$.
```

solveigvandervegt replied:
I cannot tell what the change is here, sorry, could you clarify?

rccreswell replied:
Sorry about that !
My suggestion is "later" -> "latter" in sentence 4

solveigvandervegt replied:
Ah got it, thanks!

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,268 @@
+names_rural <- names(contact_all_rural)
+names_common <- intersect(names_urban,names_rural)
+```
+We also load in all demographic data, and load in a list showing the which three-letter code for each country.
````
```suggestion
We also load in all demographic data, and load in a list showing the three-letter code for each country.
```

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,268 @@
+frac_rural <- country_rural/100 #turn percentage into fraction
+```
+
+The last step in our data prep is to create the vector of normalized population per age group for the specific country. The contact matrices have 5-year age groups with the oldest group begin 80+. Because the age demographics go up to 100+, we need to sum the last five age groups in the demographics vector to ensure compatibility with the contact matrices.
````
```suggestion
The last step in our data prep is to create the vector of normalized population per age group for the specific country. The contact matrices have 5-year age groups with the oldest group being 80+. Because the age demographics go up to 100+, we need to sum the last five age groups in the demographics vector to ensure compatibility with the contact matrices.
```

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,268 @@
+transmission_parameters(model) <- list(b, k, g, m, C)
+```
+
+Next we set the initial conditions of the model, where we specify separately what fraction of the population is infected in the urban community and what fraction is infected in the rural community. Note here that these are fractions of the total population across both communities! Because the populations are normalized, all initial conditions together must sum to one.
````
While it's obvious from the code, you could mention in the text above as well that we are starting the infection in the urban population

solveigvandervegt replied:
Good point, I have added it!

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,268 @@
+```
+
+The output from the model can then be plotted such that all populations are plotted separately for both communities
+```{r, fig.height = 4, fig.width = 7}
````
This figure could definitely be bigger for easy visibility

solveigvandervegt replied:
I have made them bigger, let me know if this is big enough!

rccreswell replied:
The size looks fine to me now

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,268 @@
+case_df <- subset(output, compartment == "Incidences_U" | compartment == "Deaths_U" | compartment == "Incidences_Y" | compartment == "Deaths_Y")
+case_df$compartment <- factor(case_df$compartment, levels = c("Incidences_U", "Deaths_U", "Incidences_Y", "Deaths_Y"))
+col <- c("Cases_U" = "grey28", "Deaths_U" = "black", "Cases_Y" = "lightgray", "Deaths_Y" = "darkgray")
+inc_plot <- ggplot(case_df, aes(x = time, y = value, fill = compartment)) +
```
I like the way you use the same general color for each compartment, but different shades for urban/rural.
However, to my eye S_Y is too light and too close in color to the background---is there a darker light blue?

If the shades end up too similar, another thing that might work is to make all the urban lines dashed, if that's possible in r.

solveigvandervegt replied:
I have chosen a slightly different shade which is hopefully a bit more blue and a bit less gray. Let me know if it is still not right!

rccreswell replied:
I think the one you picked stands out well enough against the gray, thanks

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,268 @@
+
+```
+
+The above plot shows that up to approximately $C=0.1$, the value of $C$ has a significant impact on total number of infectious individuals across the communities. Varying the value of $C$ between 0.1 and one, however, barely makes a difference. As both communities are subject to the same parameter values and the average number of contacts that individuals in both communities have don't differ a l ...
````
```suggestion
The above plot shows that up to approximately $C=0.1$, the value of $C$ has a significant impact on total number of infectious individuals across the communities. Varying the value of $C$ between 0.1 and one, however, barely makes a difference. As both communities are subject to the same parameter values and the average number of contacts that individuals in both communities have don't differ a lot in most countries, this result is not surprising. Above a certain level of interaction, the few infectious people in one community will quickly interact with individuals from the other community, spreading the infection. At that point, both communities develop their own epidemic under the same parameter values and the mixing between them further helps in synchronizing the waves of the epidemic in both communities. Only for very low levels of interaction does the initial difference in the number of infectious individuals have a lasting effect on epidemic dynamics. 
```

rccreswell reviewed (changes requested):
This looks like a very nice implementation and explanation of an interesting model---thanks @solveigvandervegt 
I gave it a first read through and have some suggestions, a few of which might help us get the tests passing

solveigvandervegt commented:
@rcw5890 Thank you for going over this so carefully!!

solveigvandervegt committed:
fixed errors

Deleted unnecessary generics from SEIRD_RU.

Incorporated comments from Richard.

solveigvandervegt committed:
fixed lint suggestions

solveigvandervegt committed:
Solved a warning

Solved the problem where documentation of function R0 in SEIRD_RU and run in SEIRDAge were incorrect, leading to warnings and improper documentation

solveigvandervegt committed:
removed aliases

removed all @aliases tags. I believe they have been phased out anyway. Removing them doesn't hurt and fixes that last warning that "checks" was throwing.

rccreswell reviewed (changes requested):
It's interesting that we are now passing on some operating systems but not on others.

Some of them seem to be complaining that your pull request adds a folder called "Data" when we already have one called "data". Presumably, you can just combine them into "data" and update the file paths.

The rgl error is more annoying. I'm guessing this happens because you are now importing matlib, which suggests rglwidget, which imports rgl (and rgl is apparently very difficult to install on macos). But, do we even need matlib? Your notebook seems to run without it, so I would try removing it as a first attempt.

solveigvandervegt committed:
remove matlib dependency

solveigvandervegt committed:
fix documentation

solveigvandervegt committed:
replace inv() with solve()

replace function inv() from matlib package with solve(), the standard R way of getting the inverse of a matrix

solveigvandervegt commented:
@rcw5890 thanks for taking another look! I fixed the matlib issue. I was using a function from that package but I found another way to do it, so the package has now been removed and the associated checks now pass. I am not sure what is going on with the error in ubuntu. I don't have a folder named Data, only data, which for as far as I can tell, is the preferred way of storing data in an R package. I thought that maybe the issue came from the fact that there is a folder named data and a file named data.R, but again, this is all following the guidelines for a proper R package. Since the other os don't complain, it must an Ubuntu issue

rccreswell commented:
Hmm, the one you have on github does seem to have both Data and data.
This is what I'm looking at:
https://github.com/Como-DTC-Collaboration/como-models/tree/i16-urban-vs-rural-communities/data
https://github.com/Como-DTC-Collaboration/como-models/tree/i16-urban-vs-rural-communities/Data

solveigvandervegt commented:
hmmmm, that's really weird cause on my local machine, I don't, so I am not sure where the Data folder is coming from!

solveigvandervegt commented:
For some reason, it appears that once I push to the origin, it is splitting all the data files into two folders. Let me see if I can change some things around

solveigvandervegt committed:
changed extensions on data files

rda is only short for rdata so doesn't change anything fundamental but maybe helps with data and Data problem?

solveigvandervegt committed:
fix typo

solveigvandervegt commented:
all fixed :)

rccreswell commented:
> all fixed :)

Thanks for making all those changes and getting the tests passing. This is looking great but I did have a couple of other various thoughts, not sure if github is hiding them?

solveigvandervegt commented:
> > all fixed :)
> 
> Thanks for making all those changes and getting the tests passing. This is looking great but I did have a couple of other various thoughts, not sure if github is hiding them?

Ah, yeah I think I just saw them! I will make the changes tomorrow!

solveigvandervegt committed:
small edits

solveigvandervegt committed:
added functionality

Allow users to set the fraction of the population that is rural and the contact rates manually, without use of data. If contact matrices are used, demographic data is required, but not vice versa.

solveigvandervegt commented:
I have made the changes you requested. There are some conditions on inputs, but hopefully those are explained sufficiently in the documentation and the vignette. I am not sure what is going on with Windows, sometimes I run the checks and it is fine, and sometimes it is not.

solveigvandervegt committed:
changes to vignette

To ensure consistency across  classes

solveigvandervegt commented:
I think it is ready, but let me know what y'all think!

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,260 @@
+```
+
+The output from the model can then be plotted such that all populations are plotted separately for both communities
+```{r, fig.height = 5, fig.width = 10}
````
Some of the text in the figures is getting cut off when I run it. 
Maybe try increasing the width, or switching to nrows=2?

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,260 @@
+```
+or such that the populations are summed across the communities and we can observe the dynamics of the susceptible, exposed, infectious, recovered and dead individuals in the whole population.
+
+```{r , fig.height = 5, fig.width = 10}
````
As above, this figure size isn't quite large enough to hold all the legend entries

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
````diff
@@ -0,0 +1,260 @@
+```
+
+## The effect of varying $C$
+The connectedness parameter $C$ indicates how connected the two communities are, i.e. how much the individuals in each are expected to interact with each other. At the extreme ends we have the communities completely connected and acting as one big community for $C=1$, and completely disconnected and acting as separate communities for $C=0$. We explore what happens for intermediate values by simul ...
````
```suggestion
The connectedness parameter $C$ indicates how connected the two communities are, i.e. how much the individuals in each are expected to interact with each other. At the extreme ends we have the communities completely connected and acting as one big community for $C=1$, and completely disconnected and acting as separate communities for $C=0$. We explore what happens for intermediate values by simulating the model for a range of values of $C$ and plotting the summed infectious population for each simulation.
```
or,  we could say something like "combined infectious population"

rccreswell commented on vignettes/SEIRD_rural_urban.Rmd:
```diff
@@ -0,0 +1,260 @@
+\end{split}
+\end{equation}
+
+\noindent In the above equations, $\phi_U$ is the fraction of the population that lives in an urban environment, $\phi_Y$ is the fraction of the population that lives in a rural environment, $N_U$ is the average number of contacts that an urban individual has in a day (normalized over the whole population), and $N_Y$ is the average number of contacts that a rural individual has in a day (normaliz ...
```
I think I see what you are doing here but I still don't see exactly what you mean by "(normalized over the whole population)" and similarly "normalized average number of contacts." Do we even need these statements, or is "average number of contacts" enough?

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,642 @@
+             BEFORE assigning the contact matrices.")
+      }
+    # compute contacts N_U and N_Y from contact matrices
+    # number of urban contacts for an urban individual, normalized over the
```
If you don't mind more of my harping on the "normalized" terminology---is there a reason why we can't just describe this as an average or weighted average?

solveigvandervegt replied:
I think my problem with that is that the populations in the model are all normalized and the units of the entries in the matrix simply aren't "number of contacts" so it seems wrong to note it as such just to simplify things. Since the number of contacts has been divided through by the total population, the numbers in the matrix are much smaller than what you would expect if you were looking at the average number of contacts, so the results we get wouldn't make sense. I understand that I need to phrase it differently cause you don't think it is clear, so I will think about it, but I do think it needs to be noted!

solveigvandervegt replied:
I think with respect to your comment above in the vignette it is different because it is a general explanation of the model without reference to our specific implementation of it.

rccreswell commented on R/SEIRD_RU.R:
```diff
@@ -0,0 +1,642 @@
+#' The differential equations for the variables in the first group are written
+#' in the form
+#' \deqn{\frac{dx_i}{dt} = F_i - V_i}
+#' where F_i are the terms describing how individuals enter the
+#' compartment and V_i the terms describing how individuals leave.
```
Thanks for adding the extra information here.
I plan to actually check your code, but right now I'm still learning the method. I have a few questions up front---
- The paper says that the derivatives should be evaluated at the disease free equilibrium, but it looks like you are evaluating at the actual initial conditions of the model. Is this allowed?
- The paper you cite says that "V_i(x) is the rate of other transitions between compartment i and other
**infected** compartments." But in most other descriptions of the method, and your implementation, V_i seems to contain transitions to non-infected compartments as well, such as R. Do you have any idea what they mean by this?
- `where F_i are the terms describing how individuals enter the compartment and V_i the terms describing how individuals leave.` I think we can specify that, in general, F_i contains only those individuals entering the compartment that are considered new infections. As for V_i, however my previous question is answered, it seems that it's not restricted to individuals leaving but can contain other transitions.

solveigvandervegt replied:
In response to your points:
1) Yes, you are correct, we need to be at a disease-free equilibrium. I have fixed this now. Not sure why I used the initial conditions before, probably because they are very, very close to the disease-free equilibrium. Thanks for catching this!
2) I have simply followed the example in the paper. If you want to do it differently, please let me know what maths you want changed and why. I am happy with how it is now.
3) done.

rccreswell reviewed (changes requested):
Thanks for making those changes
There are a few more minor things I noticed

solveigvandervegt committed:
changes requested by Richard

solveigvandervegt committed:
attempt to fix pandoc errors

solveigvandervegt committed:
fix tests

Because the computation of R0 was corrected, the test file had to be corrected as well.

solveigvandervegt committed:
Adjust comments in code

No longer refer to contact matrices as normalized, but as divided by the total population.

rccreswell reviewed (approved):
Thanks I think you've covered everything I saw
I'd be happy for @ben18785 and @IdilCazimoglu to take a look too

solveigvandervegt commented:
@rcw5890   Great, thank you so much for all your help Richard!!
